proposals/1859_github-auth/README.md

```diff
@@ -0,0 +1,527 @@
+---
+title: "Authenticating Users via GitHub"
+authors: [ "@cfryanr" ]
+status: "in-review"
+sponsor: [ ]
+approval_date: ""
+---
+
+*Disclaimer*: Proposals are point-in-time designs and decisions.
+Once approved and implemented, they become historical documents.
+If you are reading an old proposal, please be aware that the
+features described herein might have continued to evolve since.
+
+# Authenticating Users via GitHub
+
+## Problem Statement
+
+Many developers in the world have a GitHub account, and many enterprises use GitHub.
+This makes GitHub a convenient identity provider for an enterprise to use to control
+access to Kubernetes clusters for its developers.
+
+This document proposes adding a GitHubIdentityProvider resource to the Pinniped Supervisor,
+which would allow GitHub to be used as an identity provider for authentication to Kubernetes
+clusters.
+
+### How Pinniped Works Today (as of version v0.28.0)
+
+The Pinniped Supervisor currently supports OIDC-compliant identity providers and
+LDAP identity providers (including Active Directory). Unfortunately, GitHub does not offer
+either of these protocols for authenticating users. GitHub uses a slightly customized version
+of OAuth 2.0 for authenticating users, so Pinniped will need to add new features to support
+using GitHub as an identity provider.
+
+## Terminology / Concepts
+
+### GitHub OAuth 2.0 Clients
+
+For web browser-based authentication, GitHub supports two different types of OAuth 2.0 clients (compared in
+[this doc](https://docs.github.com/en/apps/creating-github-apps/about-creating-github-apps/deciding-when-to-build-a-github-app)
+and [this doc](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/differences-between-github-apps-and-oauth-apps)):
+
+- GitHub Apps (newer)
```

Comment:
`GitHub Apps` vs `GitHub OAuth Apps` is still confusing to some degree.  `Legacy GitHub Apps` perhaps?

Reply:
These are GitHub's names 💩